### PR TITLE
Load initial Create state from query param

### PR DESCRIPTION
**Pull request by tomquirk**
## What does this PR do and why?

This PR adds the ability to pre-populate project creation data from a query parameter `initialState`.

This kills a few birds:
- https://github.com/jbx-protocol/juice-interface/issues/2558. It turns out we need this for versioning. V1 projects need to launch a whole new project on V3, so this will help with that.
- Templating in future
- Sharable project config links

Here's a test link (note the project title). It's absurdly long but turns out that's probably fine (test URL below is 2093 characters long).

Besides: for when we're linking to these templates within JBM (with links and buttons and such), we can use Next magic to hide the query param in the browser URL bar anyway: https://nextjs.org/docs/api-reference/next/router#routerpush.

We could also serialize and compress this json further by tokenizing the object keys (like `projectMetadata` becomes `p`), but feels overkill for MVP.

Screenshot 2022-11-25 at 1 56 17 PM

http://localhost:3000/create?initialState=%7B%22projectMetadata%22%3A%7B%22name%22%3A%22test%20initial%20state%22%2C%22infoUri%22%3A%22%22%2C%22logoUri%22%3A%22%22%2C%22description%22%3A%22%22%2C%22twitter%22%3A%22%22%2C%22discord%22%3A%22%22%2C%22tokens%22%3A%5B%5D%2C%22version%22%3A5%7D%2C%22fundingCycleData%22%3A%7B%22duration%22%3A%220%22%2C%22weight%22%3A%221000000000000000000000000%22%2C%22discountRate%22%3A%220x00%22%2C%22ballot%22%3A%220x0000000000000000000000000000000000000000%22%7D%2C%22fundingCycleMetadata%22%3A%7B%22global%22%3A%7B%22allowSetTerminals%22%3Afalse%2C%22allowSetController%22%3Afalse%2C%22pauseTransfers%22%3Afalse%7D%2C%22reservedRate%22%3A%220x00%22%2C%22redemptionRate%22%3A%220x2710%22%2C%22ballotRedemptionRate%22%3A%220x2710%22%2C%22pausePay%22%3Afalse%2C%22pauseDistributions%22%3Afalse%2C%22pauseRedeem%22%3Afalse%2C%22allowMinting%22%3Afalse%2C%22pauseBurn%22%3Afalse%2C%22allowTerminalMigration%22%3Afalse%2C%22allowControllerMigration%22%3Afalse%2C%22holdFees%22%3Afalse%2C%22useTotalOverflowForRedemptions%22%3Afalse%2C%22useDataSourceForPay%22%3Afalse%2C%22useDataSourceForRedeem%22%3Afalse%2C%22dataSource%22%3A%220x0000000000000000000000000000000000000000%22%2C%22preferClaimedTokenOverride%22%3Afalse%2C%22metadata%22%3A%220%22%7D%2C%22fundAccessConstraints%22%3A%5B%7B%22terminal%22%3A%220x8E8d0C73ddee3819Aaa1A2943350012803Cb8AcE%22%2C%22token%22%3A%220x000000000000000000000000000000000000eeee%22%2C%22distributionLimit%22%3A%226901746346790563787434755862277025452451108972170386.555162524223799295%22%2C%22distributionLimitCurrency%22%3A%221%22%2C%22overflowAllowance%22%3A%220%22%2C%22overflowAllowanceCurrency%22%3A%220%22%7D%5D%2C%22payoutGroupedSplits%22%3A%7B%22group%22%3A1%2C%22splits%22%3A%5B%5D%7D%2C%22reservedTokensGroupedSplits%22%3A%7B%22group%22%3A2%2C%22splits%22%3A%5B%5D%7D%2C%22nftRewards%22%3A%7B%22rewardTiers%22%3A%5B%5D%2C%22collectionMetadata%22%3A%7B%22name%22%3A%22v2%20NFT%20rewards%22%2C%22description%22%3A%22NFTs%20rewarded%20to%20v2's%20Juicebox%20contributors.%22%7D%7D%2C%22mustStartAtOrAfter%22%3A%221%22%7D

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/redux/slices/editingV2Project.ts [hunk 1]
```diff
@@ -49,44 +49,39 @@ export type NftRewardsData = {
...
-  reconfigurationRuleSelection: ReconfigurationStrategy | undefined
-  createFurthestPageReached: CreatePage
-  createSoftLockPageQueue: CreatePage[] | undefined
   mustStartAtOrAfter: string
 }
 
+type ReduxState = {
+  version: number
+} & ProjectState &
+  CreateState
```
Comment (tomquirk) [inline src/redux/slices/editingV2Project.ts:76]: I think ideally we should create clearer boundaries between create-flow data and generic project data. Redux is used in project reconfiguration too, where `CreateState` properties aren't relevant. Perhaps we should even have an entirely separate redux store for the create flow.

File: src/redux/slices/editingV2Project.ts [hunk 2]
```diff
@@ -138,35 +133,58 @@ export const EMPTY_NFT_COLLECTION_METADATA = {
...
+
 const editingV2ProjectSlice = createSlice({
   name: 'editingV2Project',
-  initialState: defaultProjectState,
+  initialState: defaultReduxState,
   reducers: {
-    resetState: () => defaultProjectState,
+    setState: (_, action: PayloadAction<ReduxState>) => {
+      return action.payload
+    },
```
Comment (tomquirk) [inline src/redux/slices/editingV2Project.ts:186]: this is the only new thing.

File: src/redux/slices/editingV2Project.ts (hunk 1 above)
Comment (wraeth-eth) [reply src/redux/slices/editingV2Project.ts:76]: love this idea

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2587** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2587" target="_blank">Graphite</a> 
    * **PR #2565** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2565" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2565?utm_source=stack-comment).

**Comment by wraeth-eth** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2565?utm_source=stack-comment).


Job Reference: KTjpsSDz0BFJe3qmHJyZ

**Comment by wraeth-eth** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2565?utm_source=stack-comment).


Job Reference: P70qfCBH4OQiNOT7PvLg

**Pull request merged**